base/ltclass.dtx

```diff
@@ -1752,10 +1752,30 @@
 % \end{macro}
 %
 % \begin{macro}{\RequirePackage}
+% \changes{v1.5i}{2023/11/26}{Test group level}
 % Load package if not already loaded.
 %    \begin{macrocode}
+%</2ekernel>
+%<latexrelease>\IncludeInRelease{2024/06/01}%
+%<latexrelease>                 {\RequirePackage}{Check Group}%
+%<*2ekernel|latexrelease>
 \def\RequirePackage{%
+  \ifnum\currentgrouplevel>\z@
+     \@latex@error
+      {\noexpand\RequirePackage or \noexpand\usepackage in a group}%
+      {\noexpand\RequirePackage or \noexpand\usepackage should be at the top level}%
```

Comment:
`\noexpand\cmd` will leave an extra space after `\cmd`, `\string` won't.

If `\RequirePackageWithOptions` is needed too, perhaps
```suggestion
     \@latex@error
      {Package Loaded in a group}%
      {\string\usepackage, \string\RequirePackage, or
       \string\RequirePackageWithOptions \MessageBreak
       should be at the top level}%
```

Reply:
but you wouldn't want comma before "or" I guess

Reply:
I'd probably keep the error message shorter (but use \string)